hw-2/controllers/users.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from 'express';
+import { ValidatedRequest } from 'express-joi-validation';
+import UsersStorage from '../storage/users';
+import { UserRequestSchema, User } from '../models/user';
+
+const usersStorage = new UsersStorage();
+
+function prepareUser(user: User | null): User | null {
+  if (user === null) return null;
+
+  const preparedUser = {
+    ...user,
+  };
+  delete preparedUser.isDeleted;
```

Comment:
As a tip:

There's an `ES2015` way of "deleting" a prop:

```typescript
const { isDeleted: _, ...user: preparedUser } = user;
```

There's a V8 blog post where they kinda say that using `delete` indiscriminately might degrade performance issues... Couldn't find it to show you...

Reply:
Wow, nice tip. Was thinking how to make with destructuring. Thanks you!